### PR TITLE
Clean Up NODE_PROBLEM_DETECTOR_VERSION

### PR DESCRIPTION
**Pull request by Dragoncell**
#### What type of PR is this?
/kind cleanup 

#### What this PR does / why we need it:
Clean up the NODE_PROBLEM_DETECTOR_VERSION to avoid confusion. only DEFAULT_NPD_VERSION is needed to manage npd version 

#### Which issue(s) this PR fixes:
Fixes # None

### CODE REVIEW

**Comment by Dragoncell** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

/signed the CLA

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -305,9 +305,6 @@ if [[ "${NODE_OS_DISTRIBUTION}" == "gci" ]]; then
 else
   export ENABLE_NODE_PROBLEM_DETECTOR="${KUBE_ENABLE_NODE_PROBLEM_DETECTOR:-daemonset}"
 fi
-NODE_PROBLEM_DETECTOR_VERSION="${NODE_PROBLEM_DETECTOR_VERSION:-}"
```
Comment (SergeyKanzhelev) [inline cluster/gce/config-default.sh:308]: isn't still being used in NPD? https://github.com/kubernetes/node-problem-detector/blob/f42281ee2658900bdb0571e1159a43f6ab712a19/test/build.sh#L84 Can you please explain why it is not needed any longer?

Comment (Dragoncell) [reply cluster/gce/config-default.sh:308]: Thanks for the comment. I wasn't aware of this before. 
From my understanding, NPD used NODE_PROBLEM_DETECTOR_VERSION for new version's staging testing, therefore in k8bs's downloading logic,  NODE_PROBLEM_DETECTOR_VERSION can override DEFAULT_NPD_VERSION when it is set.  In this case, we need this logic for testing purpose, and doesn't need clean up.

Comment (SergeyKanzhelev) [reply cluster/gce/config-default.sh:308]: CC: @vteratipally

So this PR should be closed?

Comment (bobbypage) [reply cluster/gce/config-default.sh:308]: Thanks @SergeyKanzhelev for noticing the NPD e2e test dependency on NODE_PROBLEM_DETECTOR_VERSION. In that case, I think we should keep NODE_PROBLEM_DETECTOR_VERSION. We should still upgrade NPD in k/k from v0.8.5 https://github.com/kubernetes/kubernetes/blob/03709c0/cluster/gce/gci/configure.sh#L29 to latest version v0.8.7

**Comment by Dragoncell** [comment]
As discussed, the clean up is not needed, close the request.

**Pull request closed without merging**